api-resources.md

```diff
@@ -0,0 +1,414 @@
+#### Representation
+
+A paginated list of aggregation jobs for the task.
+
+```
+struct {
+  u64 page;
+  AggregateShare aggregate_shares<0..2^16-1>
+} AggregateShareList
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+The request takes an optional query parameter for the page. The response body is a `struct AggregateShareList`.
+
+### A collection
+
+Only the leader supports this resource.
+
+#### Path
+
+`/tasks/task-id/collections/collection-id`
+
+`collection-id` uniquely identifies a collection. It is chosen by the collector when it POSTs to the resource.
+
+#### Representation
+
+```
+struct {
+  PartialBatchSelector part_batch_selector;
+  uint64 report_count;
+  HpkeCiphertext encrypted_agg_shares<1..2^32-1>;
+} Collection;
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+If the collection is available, the response body is a `struct Collection`. Otherwise the server can return something like HTTP 202 Accepted to indicate it's not ready yet, or HTTP 404 if no such aggregate share request is known to the server.
+
+This is analogous to DAP-02's `GET` on a collect job URI.
+
+##### PUT
+
+PUT to a collection resource is how the collector instructs the leader to assemble the aggregate shares. The request is an idempotent PUT, because if a collection with the provided ID already exists, its representation can be returned by the helper without further state mutation.
+
+The request body is a `CollectReq`:
+
+```
+struct {
+  QueryType query_type;
+  select (Query.query_type) {
+    case time_interval: Interval batch_interval;
+    case fixed_size: Empty;
+  }
+} Query;
+
+struct {
+  Query query;
+  opaque agg_param<0..2^16-1>; /* VDAF aggregation parameter */
+} CollectReq;
+```
+
+If successful, the response from the helper is a `struct Collection`. That same share may later be obtained by a GET request on the resource.
+
+This is analogous to DAP-02's `POST /collect`.
+
+###### A note on `Query.fixed_size`
+
+`struct Query` is essentially identical to `struct BatchSelector`, so the note about `BatchSelector.fixed_size` from `PUT aggregate_share` applies here, except that a collection's ID is chosen by the collector.
+
+##### DELETE
+
+Instructs the leader to abandon the collection and allows the leader to discard all state related to it. This is akin to DELETE on an aggregate share for the helper (indeed, the leader might send a DELETE to the helper's aggregate share resource after its corresponding collection is deleted).
+
+This is analogous to DAP-02's `DELETE` on a collect job URI.
+
+#### Alternatives: `collection-id`, `PUT` vs. `POST` and idempotence?
+
+This proposal introduces the notion of a `collection-id`, which is chosen by the collector when it sends `CollectReq` to helper. In DAP-02, a collection is already uniquely identified by the combination of the aggregation parameter and the query, meaning we can get by without `collection-id`. Let's sketch out what the API would look like so we can weigh pros and cons.
```

Comment:
> a collection is already uniquely identified by the combination of the aggregation parameter and the query

nit: this is true today but won't be true once #342 is resolved; at that point, there will be some mechanism for the Collector's `Query` to say something like "I don't know the specific batch I want, just give me one that's ready (if any)" -- i.e. the Collector's `Query` will not directly identify a single specific batch.

I guess the more accurate way of saying this would be that a collection is identified by an aggregation parameter and a batch identifier (literally `BatchID` for fixed-size, a time interval for time-interval).

Reply:
A recurring set of questions here is (1) whether aggregate shares and collections need an opaque ID to uniquely identify them beyond the (aggregate parameter, query) tuple and (2) who should choose that ID if it is needed. I'm still ruminating on this; more thoughts to follow in this thread.

Reply:
- [ ] Do we need an API level concept of collection ID or aggregate share ID? If so, who chooses those IDs? Can the same solution work for both time interval and chunky DAP queries?